offchain-modules/lib/src/dapp/indexer/ckb_indexer.rs

```diff
@@ -0,0 +1,209 @@
+                    let token_addr: ETHAddress =
+                        eth_recipient.eth_token_address.get_address().into();
+                    let recipient_addr: ETHAddress =
+                        eth_recipient.eth_recipient_address.get_address().into();
+                    let token_amount = eth_recipient.token_amount;
+                    let ckb_tx_proof = parse_ckb_proof(
+                        hash.clone().as_str(),
+                        String::from(self.rpc_client.url()),
+                    )?;
+                    let proof_str = serde_json::to_string(&ckb_tx_proof)?;
+                    let record = CkbToEthRecord {
+                        ckb_burn_tx_hash: hash.clone(),
+                        status: "pending".to_string(),
+                        token_addr: Some(hex::encode(token_addr.raw_data().to_vec().as_slice())),
+                        recipient_addr: Some(hex::encode(
+                            recipient_addr.raw_data().to_vec().as_slice(),
+                        )),
+                        token_amount: Some(Uint128::from(token_amount).to_string()),
+                        ckb_spv_proof: Some(proof_str),
+                        block_number,
+                        ..Default::default()
+                    };
+                    create_ckb_to_eth_record(&self.db, &record).await?;
+                    return Ok(true);
+                }
+            }
+        }
+        Ok(false)
+    }
+
+    pub fn verify_eth_recipient_data(
+        &mut self,
+        eth_recipient: ETHRecipientDataView,
+    ) -> Result<bool> {
+        let light_client_typescript_hash_left = eth_recipient.light_client_typescript_hash;
+        let cell_script = parse_cell(
+            &self
+                .force_config
+                .deployed_contracts
+                .as_ref()
+                .unwrap()
+                .light_client_cell_script
+                .cell_script
+                .as_str(),
+        )?;
+        let mut light_client_typescript_hash = [0u8; 32];
+        light_client_typescript_hash
+            .copy_from_slice(cell_script.calc_script_hash().raw_data().as_ref());
+        let eth_bridge_lock_hash_left = eth_recipient.eth_bridge_lock_hash;
+        let mut eth_bridge_lock_code_hash = [0u8; 32];
+        eth_bridge_lock_code_hash.copy_from_slice(
+            &hex::decode(
+                &self
+                    .force_config
+                    .deployed_contracts
+                    .as_ref()
+                    .unwrap()
+                    .bridge_lockscript
+                    .code_hash,
+            )
+            .map_err(|err| anyhow!(err))?,
+        );
+        if (light_client_typescript_hash_left == light_client_typescript_hash)
+            && (eth_bridge_lock_hash_left == eth_bridge_lock_code_hash)
+        {
+            return Ok(true);
+        }
+        Ok(false)
+    }
+
+    pub async fn handle_mint_tx(&mut self, tx: Transaction) -> Result<()> {
```

Comment:
should verify the tx contains `bridge lockscript`, otherwise it can only support central-situation.

Reply:
https://github.com/LeonLi000/force-bridge-eth/commit/64230bb7555950926ba1550e8bb9d510c9dd152a#diff-d1833c71df3dc5e85e67ceefdd801c93bb23e12406de2c57a6ca02ca41edd772R205